onnxruntime/core/providers/dml/OperatorAuthorHelper/OperatorHelper.h

```diff
@@ -501,20 +534,118 @@ class SplitHelper
     std::vector<int> m_split;
 };
 
-class SliceHelper
+class SliceHelperBase
 {
 public:
+    template<typename Info_t>
     void Initialize(
-        const MLOperatorAttributes& operatorAttributes,
-        gsl::span<const DimensionType> inputDimensions
-        );
+        const Info_t& operatorInfo,
+        gsl::span<const DimensionType> inputDimensions,
+        uint32_t opsetVersion
+        )
+    {
+        std::vector<int32_t> starts;
+        std::vector<int32_t> ends;
+        std::vector<int32_t> axes;
+        std::vector<int32_t> steps;
+        if (7 == opsetVersion)
+        {
+            // Get starts, ends and axes from attributes
+            starts = operatorInfo.GetOptionalAttributeVectorInt32(AttrName::Starts);
+            ends = operatorInfo.GetOptionalAttributeVectorInt32(AttrName::Ends);
+            axes = operatorInfo.GetOptionalAttributeVectorInt32(AttrName::Axes);
+        }
+        else if (10 == opsetVersion)
+        {
+            // Get starts, ends, optional axes and optional steps from constant inputs.
+            MLOperatorTensor startsTensor = operatorInfo.GetConstantInputTensor(1);
+            const std::vector<uint32_t>& startsTensorDimensions = startsTensor.GetShape();
+            size_t dimCount = startsTensorDimensions[0];
+            const int32_t* startsData = startsTensor.GetData<int32_t>();
+            for (size_t i = 0; i < dimCount; ++i)
+            {
+                starts.push_back(gsl::narrow_cast<uint32_t>(startsData[i]));
+            }
+
+            MLOperatorTensor endsTensor = operatorInfo.GetConstantInputTensor(2);
+            const std::vector<uint32_t>& endsTensorDimensions = endsTensor.GetShape();
+            dimCount = endsTensorDimensions[0];
+            const int32_t* endsData = endsTensor.GetData<int32_t>();
+            for (size_t i = 0; i < dimCount; ++i)
+            {
+                ends.push_back(gsl::narrow_cast<uint32_t>(endsData[i]));
+            }
+            uint32_t inputCount = operatorInfo.GetInputCount();
+            if (operatorInfo.GetInputCount() > 3)
```

Comment:
Propose using `inputCount` which is other unused, rather than calling `operatorInfo.GetInputCount()` again.